iml-agent/src/daemon_plugins/ostpool.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    action_plugins::ostpool::{ost_list, pool_list, pools},
+    agent_error::ImlAgentError,
+    cmd::lctl,
+    daemon_plugins::{DaemonPlugin, Output},
+};
+use futures::{future::{join_all, try_join_all}, lock::Mutex, Future, FutureExt};
+use iml_wire_types::{OstPool, OstPoolAction};
+use std::{
+    collections::{HashMap, HashSet},
+    iter::FromIterator,
+    pin::Pin,
+    sync::Arc,
+};
+
+#[derive(Debug)]
+pub struct PoolState {
+    state: Arc<Mutex<HashMap<String, HashMap<String, HashSet<String>>>>>,
+}
+
+pub fn create() -> impl DaemonPlugin {
+    PoolState {
+        state: Arc::new(Mutex::new(HashMap::new())),
+    }
+}
+
+async fn list_fs() -> Result<Vec<String>, ImlAgentError> {
+    Ok(lctl(vec!["get_param", "-N", "mdt.*-MDT0000"])
+        .await
+        .map(|o| {
+            String::from_utf8_lossy(&o.stdout)
+                .lines()
+                .filter_map(|line| {
+                    line.split('.')
+                        .skip(1)
+                        .next()
+                        .and_then(|mdt| mdt.split("-MDT").next())
```

Comment:
I suggest making body of `filter_map` a function and writing one test for it, to make it clear what's its expected input.